power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -27,6 +28,28 @@ namespace power_grid_model::meta_data {
 constexpr auto const& as_array(msgpack::object const& obj) { return obj.via.array; }
 constexpr auto const& as_map(msgpack::object const& obj) { return obj.via.map; }
 // NOLINTEND(cppcoreguidelines-pro-type-union-access)
+
+constexpr void handle_inf(msgpack::object const& obj, auto& attr) {
+    using namespace msgpack;
+    using namespace std::string_view_literals;
+
+    auto infinity = std::numeric_limits<double>::infinity();
+    if (obj.type == type::STR) {
+        auto const& obj_string = obj.as<std::string_view>();
+        if (obj_string == "inf" || obj_string == "-inf" || obj_string == "+inf") {
```

Comment:
this is redundant. please directly check `if (... "inf" || ... "+inf") { ... } else if (... "-inf") { ... } else { throw ... }`